Add support and documentation for Windows (#410)
Cloud-CV/Fabrik: WINDOWS.md

# Windows Installaion

There are two main methods you can use to setup Fabrik on a Windows system with a virtual environment.
Firstly, you can use WSL, the Windows Subsystem for Linux, which is by far the easier option.
To install Fabrik with WSL, all you need to do is simply install a distro of your choice through the Windows Store, and then follow the Linux installation instructions in the README.
The alternative is to install natively through the Win32 subsystem, which is significantly more complicated, but an option if the WSL is not available.

### WSL Notes

> The setup is exactly the same as other linux systems. The Ubtuntu disto has been tested and confirmed to work.

Reviewer: Mention ubuntu 14.04 or 16.04

> It is iportant Fabrik is setup in `/mnt/c/...`, rather than the Linux user profile directory, because editing of the WSL subsystem from the Win32 subsystem can leading to file corruption. 

Reviewer: ```suggestion
It is important that Fabrik is setup in `/mnt/c/...`, rather than the Linux user profile directory, because editing of the WSL subsystem from the Win32 subsystem can leading to file corruption. 
```
Minor typo

    [This archive](https://github.com/ServiceStack/redis-windows/raw/master/downloads/redis-latest.zip)
    has been tested and appears to work, but you may want to search for better alternatives. To use, simply
    extract and run `redis-server.exe redis.windows.conf`. A firewall prompt may appear, be sure to click 'Allow'.
    
    On line 115 of `settings\common.py` set the hostname to localhost, like so:
    ```
    "CONFIG": {
        # replace redis hostname to localhost if running on local system
        "hosts": [("localhost", 6379)],
        "prefix": u'fabrik:',
    },
    ```
    In the same file ensure the celery result backend around line 122 is set to the local redis server
    ```
    CELERY_RESULT_BACKEND = 'redis://localhost:6379/0'
    ```
    Ensure that line 8 of ` ide\celery_app.py ` has both `broker` and `backend` parameters set to `'redis://localhost:6379/0'`. It should look like:
    ```
    app = Celery('app', broker='redis://localhost:6379/0' backend='redis://localhost:6379/0', include=['ide.tasks'])
    ```

5. Install Tensorflow, Keras and Caffe. Note that this will fail if the wrong version of Python is used.
    

>     Tenserflow (this will fail if you don't have either python 3.6, 3.5 or 3.4 64 bit):

Reviewer: ```suggestion
- Tensorflow (this will fail if you don't have either python 3.6, 3.5 or 3.4 64 bit):
```
Minor typo

    ```
    (venv) cmd> pip install tensorflow==1.4
    ```

>     Keras:

Reviewer: ```suggestion
- Keras:
```

    ```
    (venv) cmd> pip install keras==2.0.8
    ```

>     A work-in-progess Windows port of Caffe is currently maintaiend 

Reviewer: ```suggestion
- A work-in-progess Windows port of Caffe is currently maintaiend 
```

    and links to prebuilt binaries are listed in the readme. However, at the time of writing,
    the downloads were broken, so in order to obtain the binaries, you need to download them
    manually. Go to the [AppVeyor CI for Caffe](https://ci.appveyor.com/project/BVLC/caffe/branch/windows)
    and find the latest build. Pick the desired environment and download and extract the generated artifact.
    Run the command (you'll need to do this every time you start a new session)
    ```
    (venv) cmd> SET PYTHONPATH=%PYTHONPATH%;{put location of caffe installaion here}\python;
    ```
    Open the `requirements.txt` of the caffe installation and comment out the `leveldb` line. Run the following command.
    ```
    (venv) cmd> pip install -r "{put location of caffe installaion here}\python\requirements.txt"
    pip install --upgrade python-dateutil
    ```
    Edit `settings\common.py` and change `CAFFE_TOOLS_DIR` on line 124 to `{put location of caffe installaion here}\bin`.

6. Install other dependencies. Edit `requirements\common.txt` line 8 and change `tensorflow==1.4.1` to `tensorflow==1.4`.

>     Comment out the `uWSGI` line, as it is incompatible with windows and is only needed for depoyment.

Reviewer: ```suggestion
    Comment out the `uWSGI` line, as it is incompatible with windows and is only needed for deployment.
```
Minor typo

>     Install Microsoft Visual C++ Tools for the Twisted package by installing following https://visualstudio.microsoft.com/visual-cpp-build-tools/. Install

Reviewer: ```suggestion
    Install Microsoft Visual C++ Tools for the Twisted package by installing the [following](https://visualstudio.microsoft.com/visual-cpp-build-tools/). Install
```
Adding markdown
